larpandoracontent/LArHelpers/LArMCParticleHelper.h

```diff
@@ -190,29 +239,33 @@ class LArMCParticleHelper
      *  @param  parameters validation parameters to decide when an MCParticle is considered reconstructable
      *  @param  fCriteria a function which returns a bool (= shouldSelect) for a given input MCParticle
      *  @param  selectedMCParticlesToHitsMap the output mapping from selected mcparticles to their hits
+     *  @param  testBeamHierarchyMode test beam hierarchy mode
      */
     static void SelectReconstructableMCParticles(const pandora::MCParticleList *pMCParticleList, const pandora::CaloHitList *pCaloHitList,
-        const PrimaryParameters &parameters, std::function<bool(const pandora::MCParticle *const)> fCriteria, MCContributionMap &selectedMCParticlesToHitsMap);
+        const PrimaryParameters &parameters, std::function<bool(const pandora::MCParticle *const)> fCriteria, MCContributionMap &selectedMCParticlesToHitsMap,
+        const bool testBeamHierarchyMode = false);
 
     /**
      *  @brief  Get mapping from Pfo to reconstructable 2D hits (=good hits belonging to a selected reconstructable MCParticle)
      *
      *  @param  pfoList the input list of Pfos
      *  @param  selectedMCParticleToHitsMap the input mapping from selected reconstructable MCParticles to their hits
      *  @param  pfoToReconstructable2DHitsMap the output mapping from Pfos to their reconstructable 2D hits
+     *  @param  testBeamHierarchyMode test beam hierarchy mode
      */
     static void GetPfoToReconstructable2DHitsMap(const pandora::PfoList &pfoList, const MCContributionMap &selectedMCParticleToHitsMap,
-        PfoContributionMap &pfoToReconstructable2DHitsMap);
+        PfoContributionMap &pfoToReconstructable2DHitsMap, const bool testBeamHierarchyMode = false);
```

Comment:
Anything we can do to push logic into the algorithms, and then maybe have two functions, or one function that calls a second function for extra processing when in test beam hierarchy mode, would help. This is as opposed to a Boolean in the function itself. This would fit with the idea of having clean helper functions that can be reviewed in isolation, together with separate algorithms that drive particular use cases.